kong/plugins/zipkin/handler.lua

```diff
@@ -45,6 +45,17 @@ local function get_reporter(conf)
 end
 
 
+local function tag_with_durations(span)
+  ctx = ngx.ctx
+  span:set_tag("kong.rewrite.duration_ms", ctx.KONG_REWRITE_TIME or 0)
+  span:set_tag("kong.preread.duration_ms", ctx.KONG_PREREAD_TIME or 0)
+  span:set_tag("kong.access.duration_ms", ctx.KONG_ACCESS_TIME or 0)
+  span:set_tag("kong.balancer.duration_ms", ctx.KONG_BALANCER_TIME or 0)
+  span:set_tag("kong.header_filter.duration_ms", ctx.KONG_HEADER_FILTER_TIME or 0)
+  span:set_tag("kong.body_filter.duration_ms", ctx.KONG_BODY_FILTER_TIME or 0)
```

Comment:
Some phases are not available in stream mode, better not to include spans with an empty value.